modules/apps/knowledge-base/knowledge-base-api/src/main/java/com/liferay/knowledge/base/exception/LockedKBArticleException.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.knowledge.base.exception;
+
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.lock.DuplicateLockException;
+import com.liferay.portal.kernel.lock.Lock;
+
+/**
+ * @author Marco Galluzzi
+ */
+public class LockedKBArticleException extends PortalException {
+
+	public LockedKBArticleException() {
+	}
+
+	public LockedKBArticleException(String msg) {
+		super(msg);
+	}
+
+	public LockedKBArticleException(String msg, Throwable throwable) {
+		super(msg, throwable);
+
+		_setLock(throwable);
+	}
+
+	public LockedKBArticleException(Throwable throwable) {
+		super(throwable);
+
+		_setLock(throwable);
+	}
+
+	public String getActionCommand() {
+		return _actionCommand;
+	}
+
+	public String getActionURL() {
+		return _actionURL;
+	}
+
+	public Lock getLock() {
+		return _lock;
+	}
+
+	public String getUserName() {
+		if (_lock != null) {
+			return _lock.getUserName();
+		}
+
+		return null;
+	}
+
+	public void setActionCommand(String actionCommand) {
+		_actionCommand = actionCommand;
+	}
+
+	public void setActionURL(String actionURL) {
+		_actionURL = actionURL;
+	}
+
+	private void _setLock(Throwable throwable) {
+		if (throwable instanceof DuplicateLockException) {
+			DuplicateLockException duplicateLockException =
+				(DuplicateLockException)throwable;
+
+			_lock = duplicateLockException.getLock();
+		}
+	}
```

Comment:
This method looks like a setter, but it doesn't follow JavaBeans conventions (`setFoo(Foo foo)`).

To avoid the type mismatch, why not create a `_getLock(Throwable)` to extract the lock from the exn, and set the value in each constructor?
```

	public LockedKBArticleException(String msg, Throwable throwable) {
		super(msg, throwable);

		_lock = _getLock(throwable);
	}
```

Reply:
Much better, thanks.

Reply:
Change done, see [d491ab3](https://github.com/liferay-content-management/liferay-portal/pull/5187/commits/d491ab3ff5d85de0e381888bd69e1b6fecb66155)